libavformat/dashenc.c

```diff
@@ -2013,11 +2017,32 @@ static int dash_write_packet(AVFormatContext *s, AVPacket *pkt)
         seg_end_duration = c->seg_duration;
     }
 
-#if 0
+    if (s->prev_pts != AV_NOPTS_VALUE) {
+        int64_t delta = pkt->pts - s->prev_pts;
+        delta = delta > 0 ? delta : (-1)*delta;
+        if (delta < s->min_pts_jump)
+            s->min_pts_jump = delta;
+        if (delta > s->max_pts_jump)
+            s->max_pts_jump = delta;
+    }
+
+    int64_t frame_duration_variation = 0;
+
+    if (s->prev_pts != AV_NOPTS_VALUE)
+        frame_duration_variation = (s->max_pts_jump - s->min_pts_jump) * 2;
```

Comment:
Shouldn't frame variation just be max jump?

Reply:
No, imagine all the frames duration are the same and we jump 1000 pts every time. Then max_pts_jump would be equal to min_pts_jump and the difference is 0.
The other case is when the frame duration is not constant and frame pts jumps sometime 272 and sometimes 256. In this case max_pts_jump would be 272 and min_pts_jump would be 256 and the frame_duration_variant = (272-256)*2 = 32